### DISCUSSION TITLE
Telecomando tv sony bravia

### DISCUSSION THREAD

**Discussion by Olodumdomotico** [indomus/forum#246] [HUB: Home Assistant]
Salve ho versione HASSIO  su raspberry, acquistato da poco broadlink ed installato relativa integrazione.
Tramite HACS ho installato smartIR , ma non ho eseguito alcuna configurazione ulteriore.

Attualmente ho una bravia smart e quindi connessa e già visibile da HA e tramite questa scheda in panoramica in teoria potrei controllare TV, ma risulta veramente proibitivo
[![Immagine-2021-05-01-172603.png](https://i.postimg.cc/sgL8gb2k/Immagine-2021-05-01-172603.png)](https://postimg.cc/K1nQ9qBf)

Però non sono capace a creare nè interfaccia a telecomando nè smartIR etc...... , ma in teoria vorrei riuscire a creare qualcosa del genere e sfruttare il codice IR del broadlink
[![1a5f43b5ad17cbd1c7851fb6580d87096c2c555c-2-562x500.png](https://i.postimg.cc/Vv4ZQVTH/1a5f43b5ad17cbd1c7851fb6580d87096c2c555c-2-562x500.png)](https://postimg.cc/jnw4yZRz)

[![telecomandi-virtuali.jpg](https://i.postimg.cc/m226NT1x/telecomandi-virtuali.jpg)](https://postimg.cc/ThF9TMcQ)

**Comment by snakuzzo** [comment]
Lasciando stare l'interfaccia telecomando (che io trovo una cosa brutta e inutile e non mi metterei nemmeno 5 min per vedere come fare), ma se hai una tv smart che espone API (anche io ho un Bravia), che senso ha usare il Broadlink? Usa direttamente le API

**Reply by Olodumdomotico** [reply]
Non saprei fare neanche questo :)    (ora vedo se trovo una guida giusto per infarinatura base)
Se puoi spiegarmi un pochino il discorso API , volentieri.....quando leggendo i forum e dicono di andare in config.yaml già mi perdo e video tutorial per mega nabbi non li ho trovati

In teoria stavo provando i comandi IR in maniera da prendere confidenza e poi fare il condizionatore che non è smart

**Reply by snakuzzo** [reply]
se non sai nemmeno di cosa si sta parlando, credo sia inutile fare come ti ho detto. Ci vuole un minimo di conoscenza e non c'è "il manuale di istruzioni". Internet è pieno di fonti per studiare...basta usare un motore di ricerca

**Reply by Olodumdomotico** [reply]
sto cercando di mettere insieme le info proprio guardando più fonti
non mi è chiaro se devo copiare dei file tramite FILEZILLA in client o tramite addon "file editor"
Non riesco a reperire i comandi e trasformarli in tasti visto che ho già gli API disponibili

Capisco che serve pazienza, ma anche piccoli aiuti o semplici link sono apprezzati
Grazie

**Reply by Olodumdomotico** [reply]
Al momento sono riuscito a domotizzare il climatizzatore tramite smartIR anche perchè con un pò di fortuna ho potuto sfruttare un "device_code" funzionante. 

Tornando al progetto base, come posso sfruttare le API del mio TV ?? mi servirebbe una scheda in panoramica più completa dell'attuale media player che è troppo compatto.
Suggerimenti??

**Reply by snakuzzo** [reply]
O ti studi il codice sorgente di HA sull'integrazione Bravia oppure studi le API...
https://pro-bravia.sony.net/develop/integrate/rest-api/spec/

**Reply by Olodumdomotico** [reply]
chissà magari mi sono imbarcato in questione ardua per le mie conoscenze.
Cliccando in "strumenti per lo sviluppatore" all'entità della TV "media_player.sony_bravia_tv"  di lato ci sono tutti gli "attributi" che essa può compiere e speravo di assegnare quest'ultime a tasti virtuali.

Vorrei evitare di scansionare a mano tutti i tasti in windows e poi trascriverli in button card personalizzati
Comunque grazie

**Reply by Olodumdomotico** [reply]
Sto seguendo un pochino meglio la guida su indomus [https://indomus.it/guide/gestire-telecomandi-virtuali-su-home-assistant-via-broadlink-parte-1/](url) ed ho fatto la prova di acquisizione del tasto on/off dall'interfaccia servizi .....credo sia andato a buon fine
Però non capisco dove prendere il codice generato

**Reply by Olodumdomotico** [reply]
[![Immagine-2021-05-04-224248.png](https://i.postimg.cc/76FDZYPJ/Immagine-2021-05-04-224248.png)](https://postimg.cc/zbF4dNgq)
Tramite filezilla vedo i 2 file creati (provati 2 tasti) , ma poi mi perdo un pochino sul da fare  :)

**Comment by Olodumdomotico** [comment]
Al momento sono bloccato a questo punto, tramite la pagina "servizi" in strumenti per sviluppatori uso questa breve compilazione:

entity_id: remote.broadlink_rm4_pro_remote 
 device: TV
 command: on_off  (questo lo vario ad ogni acquisizione)
 alternative: false  (per la tv ho solo tasti singoli ad eccezione on_off)
 timeout: 30

Dopo di che il broadlink lampeggia in attesa del segnale , premo tasto sul telecomando originale e nella cartella /config/.storage si crea un file simile a questo broadlink_remote_MAC_ADDRESS_codes
A questo punto non so come trasformare l'acquisizione in un tasto per nell'interfaccia  "panoramica"

**Reply by KC-inDomus** [reply]
segui con attenzione questa guida:

https://indomus.it/guide/gestire-telecomandi-virtuali-su-home-assistant-via-broadlink-parte-1/

**Reply by Olodumdomotico** [reply]
E' la guida che sto seguendo, però qualcosa mi sfugge.
Per vedere la cartella ".storage" va bene usare filezilla??  devo aprire file in qualche maniera per vederlo in JSON ??
In "servizi" se compilo in questa maniera

service: remote.send_command
data:
    entity_id: remote.broadlink_rm4_pro_remote
    device: TV
    command: on_off

Mi dice:  chiamata fallita a servizio remote.send_command

**Reply by Olodumdomotico** [reply]
Per adesso sono riuscito a far funzionare correttamente anche la funzione SEND  lol
Ora ho provato a creare un tasto virtuale a caso in panoramica

type: button
tap_action:
  action: call-service
  service: remote.send_command
  service_data:
    entity_id: remote.broadlink_rm4_pro_remote
    device: tv
    command: 2

Però quando lo clicco mi dice command 2 no found , però da servizi faccio il test e funziona, cosa sto sbagliando nella compilazione yaml ??

**Reply by Olodumdomotico** [reply]
Scusate se riesumo a distanza di tempo, ma il lavoro mi ha tenuto lontano dai passatempi :) 
Riepilogando se chiamo da "servizi" tramite questo comando funziona tutto 

service: remote.send_command
data:
  entity_id: remote.broadlink
  device: TV
  command: 2

Invece se aggiungo un tasto virtuale in "panoramica" mi dice comando non trovato, mi aiutate a creare compilazione corretta??

type: 'custom:button-card'
color_type: card
tap_action:
  action: call-service
  service: remote.send_command
  service_data:
    entity_id: remote.broadlink
    device: tv
    command: 2
icon: 'mdi:numeric-2'
color: 'rgb(10, 170, 238)'

**Reply by snakuzzo** [reply]
Quando posti blocchi di codice devi usare il markdown. Leggi il regolamento.
Controlla la documentazione della custom card [QUI](https://github.com/custom-cards/button-card)

```yaml
- type: 'custom:button-card'
  entity: input_select.cube_mode
  tap_action:
    action: call-service
    service: remote.send_command
    service_data:
      entity_id: remote.broadlink
      device: tv
      command: 2
```

**Reply by Olodumdomotico** [reply]
Ok, scusate provo a sciverlo in maniera corretta e leggibile in maniera da capire meglio i miei eventuali errori
Questo è il comando da "service" che funziona correttamente

```
service: remote.send_command
data:
  entity_id: remote.broadlink
  device: TV
  command: 5
```

Invece da panoramica creo il tasto e viene visualizzato correttamente, ma non esegue il comando
Ovviamente non guardate l'estetica , è solo una prova ........devo far funzionare il tasto poi creerò un layout 


[![Immagine-2021-05-22-223105.png](https://i.postimg.cc/ZKSddkVJ/Immagine-2021-05-22-223105.png)](https://postimg.cc/z3xBscsQ)

**Reply by snakuzzo** [reply]
l'errore che ti dà ? Guarda il log completo

**Reply by Olodumdomotico** [reply]
la scritta completa che visualizza è
"Fallita chiamata a servizio remore/send_command . Command not found '2'

**Reply by Olodumdomotico** [reply]
I'acquisizione è ok (learnig da servizio)
L'invio dei comandi è ok (send da servizio)
L'invio dei comandi da interfeccia lovelace guardando alcune guide mi sembra corretta , però non riesce
Avete qualche suggerimento o prove da effettuare??? grazie !!
